code/game/dna/genes/vg_powers.dm

```diff
@@ -242,6 +252,87 @@
 		for(var/mob/dead/observer/G in GLOB.player_list)
 			G.show_message("<i>Telepathic message from <b>[user]</b> ([ghost_follow_link(user, ghost=G)]) to <b>[target]</b> ([ghost_follow_link(target, ghost=G)]): [say]</i>")
 
+/obj/effect/proc_holder/spell/targeted/mindscan
+	name = "Scan Mind"
+	desc = "Offer people a chance to share their thoughts!"
+	charge_max = 0
+	clothes_req = 0
+	stat_allowed = 0
+	invocation_type = "none"
+	range = -2
+	selection_type = "range"
+	action_icon_state = "genetic_mindscan"
+	var/list/available_targets = list()
+
+/obj/effect/proc_holder/spell/targeted/mindscan/choose_targets(mob/user = usr)
+	var/list/targets = new /list()
+	var/list/validtargets = new /list()
+	var/turf/T = get_turf(user)
+	for(var/mob/living/M in range(14, T))
+		if(M && M.mind)
+			if(M == user)
+				continue
+			validtargets += M
+
+	if(!validtargets.len)
+		to_chat(user, "<span class='warning'>There are no valid targets!</span>")
+		start_recharge()
+		return
+
+	targets += input("Choose the target to listen to.", "Targeting") as null|mob in validtargets
```

Comment:
Keep in mind that the user will be able to keep this window open for a long time, and select a target after they have left the 12-tile radius. Is this as intended?